Sources/TuistGenerator/Utils/SettingsHelper.swift

```diff
@@ -44,4 +38,28 @@ final class SettingsHelper {
     func variant(_ buildConfiguration: BuildConfiguration) -> BuildSettingsProvider.Variant {
         return buildConfiguration.variant == .debug ? .debug : .release
     }
+
+    // MARK: - Private
+
+    private func resolveValue(oldValue: SettingValue?, newValue: SettingValue) -> SettingValue {
+        guard let oldValue = oldValue else {
+            return newValue
+        }
+        guard oldValue != newValue else {
+            return oldValue
+        }
+
+        switch (oldValue, newValue) {
+        case let (.string(old), .string(new)) where new.contains("$(inherited)"):
+            return .array([old, new])
+        case let (.string(old), .array(new)) where new.contains("$(inherited)"):
+            return .array([old] + new)
+        case let (.array(old), .string(new)) where new.contains("$(inherited)"):
+            return .array(old + [new])
+        case let (.array(old), .array(new)) where new.contains("$(inherited)"):
+            return .array(old + new)
+        default:
```

Comment:
Can you explain this logic? I'm struggling to follow. Why are we checking for `$(inherited)` here?

Reply:
Yes, it's a tricky bit. I was trying to preserve the behavior we currently have (but it's not 100% the same).

1. assign new value when old value is nil (`buildSettings[key] == nil` ) (covered by the first `guard` statement)
2. assign new value when it's a string and does not contain "$(inherited)" (covered by `switch default:`)
3. if both old and new are strings and both != nil, and second contains `"$(inherited)"`, then concatenate them (covered by the first `case let` statement).

I've changed that so we use the same logic while merging arrays or an array and a string, so once we update 

```
buildSettings["LD_RUNPATH_SEARCH_PATHS"] = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks"
```

to 

```
  buildSettings["LD_RUNPATH_SEARCH_PATHS"] = ["$(inherited)", "@executable_path/Frameworks", "@loader_path/Frameworks"]
```

we can use 

```
customSettings["LD_RUNPATH_SEARCH_PATHS"] = ["$(inherited)", "newValue"]
```

to extend the setting. 

**However, both string and array concatenation have one common problem:**
When we extend something that already contains `"$(inherited)"` with a new value that also contains `"$(inherited)"`, in the result we end up with two `"$(inherited)"` values.

```
{
        // Given
        settings["A"] = ["$(inherited)", "A_VALUE"]

        // When
        subject.extend(buildSettings: &settings, with: ["A": ["$(inherited)", "A_VALUE_2", "A_VALUE_3"]])

        // Then
        XCTAssertEqual(settings, ["A": ["$(inherited)", "A_VALUE", "$(inherited)", "A_VALUE_2", "A_VALUE_3"]])
}
```

I am tempted to convert the array (values) to Set to remove the duplications (and sort). For such a small data sets it *shouldn't* make a visible performance impact. 
We can also deal with strings in a similar way.

Old implementation for reference:
```
other.forEach { key, value in
    if buildSettings[key] == nil || (value as? String)?.contains("$(inherited)") == false {
        buildSettings[key] = value
    } else if let previousValueString = buildSettings[key] as? String, let newValueString = value as? String,
        previousValueString != newValueString {
        buildSettings[key] = "\(previousValueString) \(newValueString)"
    } else {
        buildSettings[key] = value
    }
}
```